src/frontend/app/features/service-catalog/add-service-instance/specify-details-step/specify-details-step.component.ts

```diff
@@ -0,0 +1,239 @@
+
+    this.allServiceInstances$ = this.initServiceInstances(paginationKey);
+
+    this.spaces$ = this.initSpacesObservable();
+  }
+
+  setOrg = (guid) => this.store.dispatch(new SetCreateServiceInstanceOrg(guid));
+
+  initServiceInstances = (paginationKey: string) => getPaginationObservables<APIResource<IServiceInstance>>({
+    store: this.store,
+    action: new GetServiceInstances(this.servicesService.cfGuid, paginationKey),
+    paginationMonitor: this.paginationMonitorFactory.create(
+      paginationKey,
+      entityFactory(serviceInstancesSchemaKey)
+    )
+  }, true)
+    .entities$.pipe(
+    share(),
+    first()
+    )
+  ngOnDestroy(): void {
+    this.orgSubscription.unsubscribe();
+    this.serviceInstanceNameSub.unsubscribe();
+  }
+
+  initOrgsObservable = (action) => getPaginationObservables<APIResource<IOrganization>>({
+    store: this.store,
+    action: action,
+    paginationMonitor: this.paginationMonitorFactory.create(
+      action.paginationKey,
+      entityFactory(organizationSchemaKey)
+    )
+  }, true)
+    .entities$.pipe(
+    share(),
+    first()
+    )
+
+  ngAfterContentInit() {
+    this.validate = this.stepperForm.statusChanges
+      .map(() => {
+        return this.stepperForm.valid;
+      });
+
+    this.orgSubscription = this.orgs$.pipe(
+      tap(o => {
+        const orgWithSpaces = o.filter(org => org.entity.spaces.length > 0);
+        if (orgWithSpaces.length > 0) {
+          const selectedOrgId = orgWithSpaces[0].metadata.guid;
+          this.stepperForm.controls.org.setValue(selectedOrgId);
+          this.store.dispatch(new SetCreateServiceInstanceOrg(selectedOrgId));
+        }
+      })
+    ).subscribe();
+
+    this.updateServiceInstanceNames();
+  }
+
+  initSpacesObservable = () => this.store.select(selectOrgGuid).pipe(
+    filter(p => !!p),
+    combineLatest(this.orgs$),
+    map(([guid, orgs]) => {
+      const filteredOrgs = orgs.filter(org => org.metadata.guid === guid);
+      return filteredOrgs.length > 0 ? filteredOrgs[0] : null;
+    }),
+    filter(p => !!p),
+    map(org => org.entity.spaces),
+    tap(spaces => {
+      if (spaces.length > 0) {
+        const selectedSpaceId = spaces[0].metadata.guid;
+        this.stepperForm.controls.space.setValue(selectedSpaceId);
+        this.store.dispatch(new SetCreateServiceInstanceSpace(selectedSpaceId));
+      }
+    })
+  )
+
+  updateServiceInstanceNames = () => {
+    this.serviceInstanceNameSub = this.stepperForm.controls.space.statusChanges.pipe(
+      combineLatest(this.allServiceInstances$),
+      map(([c, services]) => {
+        return services.filter(s => s.entity.space_guid === this.stepperForm.controls.space.value);
+      }),
+      tap(o => {
+        this.allServiceInstanceNames = o.map(s => s.entity.name);
+      })
+    ).subscribe();
+  }
+
+  onNext = () => {
+    return this.store.select(selectServicePlan).pipe(
```

Comment:
Hit a couple of issues when supply the service json param. Supplied `{ "TEST" : true }` and got the following error (using the new error PR backend #2085) 
```
code:1001
description:"Request invalid due to parse error: Field: parameters, Error: Expected instance of Hash, given an instance of String"
error_code:"CF-MessageParseError"
```
1) Is ok content for that field? If not, the JSON validation would really be helpful here
2) The stepper continued onto the next step, didn't look like the error was being set correctly. Think this might go away when the PR above is merged but worth trying out indepentently